server/events.go

```diff
@@ -267,6 +267,11 @@ func (s *Server) internalSendLoop(wg *sync.WaitGroup) {
 			c.pa.reply = []byte(pm.rply)
 			c.mu.Unlock()
 
+			if c.trace {
```

Comment:
wonder if we should collect c.trace, c.pa.subject, etc.. while under the lock to prevent data races.. or simply move the tracing under the lock.

Reply:
As of now, a clients trace value does not change. 

I filed a separate issue and for this.